[#16123] Handle server failures during post-start

jabolina opened the pull request:
* Handle failures during post-start and disable the connector.
* Simulate failure with missing binaries.

Now the initialization also handles and logs any failures during the post-start phase.
Closes #16123.

jabolina commented:
The failure is related.

jabolina commented:
Fixed. The test was expecting a CacheConfigurationException from the starting the Hot Rod server and I was throwing a CacheException. I've changed the exception to CacheConfigurationException.

wburns commented on server/runtime/src/main/java/org/infinispan/server/Server.java:
```diff
@@ -508,7 +509,7 @@ public synchronized CompletableFuture<ExitStatus> run() {
             );
          }
-         cacheManagerStart.whenComplete((ignore, t) -> {
+         initialization = cacheManagerStart.whenComplete((ignore, t) -> {
```
This seems unneeded. Can we not just widen the try/catch inside the `whenComplete` lambda that catches Throwable which just invokes `r.completeExceptionally(t);` ? And maybe change the try/catch of the sync code path to be Throwable as well.

wburns commented on server/tests/src/test/java/org/infinispan/server/resilience/NativeMissingResilienceIT.java:
```diff
@@ -0,0 +1,192 @@
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 10.0
```
Wrong tags here :)

tristantarrant commented on server/runtime/src/main/java/org/infinispan/server/SecurityActions.java:
```diff
@@ -65,7 +65,19 @@ static void startProtocolServer(ProtocolServer<ProtocolServerConfiguration> serv
+
+               try {
+                  server.stop();
+               } catch (Throwable t2) {
```
I don't think we want to start the server at all if there is a failure. I would probably not even bother stopping the protocol and exit immediately

jabolina commented:
Updated. Failures during the post-start process will be logged, and the server will stop. Updated so that missing the Lua engine does not crash the server. Instead, the respCache works without the script feature enabled.

jabolina commented:
Done a proper rebase now.
